src/DeploymentTemplate.ts

```diff
@@ -49,19 +44,15 @@ export class DeploymentTemplate {
     private _topLevelVariableDefinitions: CachedValue<IVariableDefinition[]> = new CachedValue<IVariableDefinition[]>();
     private _topLevelParameterDefinitions: CachedValue<ParameterDefinition[]> = new CachedValue<ParameterDefinition[]>();
 
-    private _schema: CachedValue<Json.StringValue | undefined> = new CachedValue<Json.StringValue | undefined>();
-
     /**
      * Create a new DeploymentTemplate object.
      *
      * @param _documentText The string text of the document.
      * @param _documentId A unique identifier for this document. Usually this will be a URI to the document.
      */
-    constructor(private _documentText: string, private _documentId: string) {
-        nonNullOrEmptyValue(_documentId, "_documentId");
-
-        this._jsonParseResult = Json.parse(_documentText);
-        this._topLevelValue = Json.asObjectValue(this._jsonParseResult.value);
+    constructor(documentText: string, documentId: string) {
+        super(documentText, documentId);
+        nonNullOrEmptyValue(documentId, "documentId");
```

Comment:
>nonNullOrEmptyValue(documentId, "documentId"); [](start = 8, length = 46)

Should this be done in the base, or is it OK for the base to have null documentId?

Reply:
OK, it is now.

---
In reply to: [407829967](https://github.com/microsoft/vscode-azurearmtools/pull/580#discussion_r407829967) [](ancestors = 407829967)